Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/type/CoronaTestProcessor.kt

```diff
@@ -0,0 +1,14 @@
+package de.rki.coronawarnapp.coronatest.type
+
+import de.rki.coronawarnapp.coronatest.qrcode.CoronaTestQRCode
+
+interface CoronaTestProcessor {
+
+    val type: CoronaTest.Type
+
+    suspend fun create(data: CoronaTestQRCode): CoronaTest
+
+    suspend fun markSubmitted(test: CoronaTest): CoronaTest
+
+    suspend fun pollServer(test: CoronaTest): CoronaTest
+}
```

Comment:
Can we use generics ? 
```kotlin 
interface CoronaTestProcessor<C : CoronaTestQRCode, T : CoronaTest> {

    val type: CoronaTest.Type

    suspend fun create(data: C): T

    suspend fun markSubmitted(test: T): T

    suspend fun pollServer(test: T): T
}
```

and then in derived Processors we won't need to cast:

```kotlin 
class PCRProcessor @Inject constructor(
    private val timeStamper: TimeStamper,
    private val verificationServer: VerificationServer,
) : CoronaTestProcessor<CoronaTestQRCode.PCR, PCRCoronaTest> {

    override val type: CoronaTest.Type = CoronaTest.Type.PCR

    override suspend fun create(data: CoronaTestQRCode.PCR): PCRCoronaTest {
    }

    override suspend fun markSubmitted(test: PCRCoronaTest): PCRCoronaTest {
       
    }

    override suspend fun pollServer(test: PCRCoronaTest): PCRCoronaTest {
    }
}
```

Reply:
That was my first implementation :smile:. I tried that for an hour and gave up. I didn't manage to get the magic combination of `@JvmSupressWildcards` working. You are very welcome to try it, I'm very curious to know what I overlooked when trying it.